rskj-core/src/test/java/co/rsk/peg/BridgeSerializationUtilsTest.java

```diff
@@ -1019,38 +1023,122 @@ public void deserializeInteger() {
         Assert.assertEquals(1200, BridgeSerializationUtils.deserializeInteger(RLP.encodeBigInteger(BigInteger.valueOf(1200))).intValue());
     }
 
+    @Test
+    public void deserializeKeccak256() {
+        String expectedValueHex = "0200000000000000000000000000000000000000000000000000000000000000";
+        String serializedValueHex = "a00200000000000000000000000000000000000000000000000000000000000000";
+        byte[] serializedValueBytes = HexUtils.decode(serializedValueHex.getBytes());
+
+        Optional<Keccak256> result = BridgeSerializationUtils.deserializeKeccak256(serializedValueBytes);
+        Assert.assertTrue(result.isPresent());
+
+        Keccak256 expectedHash = new Keccak256(HexUtils.decode(expectedValueHex.getBytes()));
+        Assert.assertEquals(expectedHash, result.get());
+    }
+
+    @Test
+    public void deserializeKeccak256_nullValue() {
+        Optional<Keccak256> result = BridgeSerializationUtils.deserializeKeccak256(null);
+        Assert.assertFalse(result.isPresent());
+    }
+
+    @Test
+    public void deserializeKeccak256_zeroHash() {
+        String expectedHashHex = "0000000000000000000000000000000000000000000000000000000000000000";
+        String expectedSerializedHashHex = "a00000000000000000000000000000000000000000000000000000000000000000";
+        byte[] expectedSerializedHashBytes = HexUtils.decode(expectedSerializedHashHex.getBytes());
+
+        Optional<Keccak256> result = BridgeSerializationUtils.deserializeKeccak256(expectedSerializedHashBytes);
+        Assert.assertTrue(result.isPresent());
+
+        Keccak256 expectedHash = new Keccak256(HexUtils.decode(expectedHashHex.getBytes()));
+        Assert.assertEquals(expectedHash, result.get());
+    }
+
+    @Test
+    public void serializeKeccak256() {
+        Keccak256 originalHash = PegTestUtils.createHash3(2);
```

Comment:
might be worth leaving a comment stating that calling `createHash3(2)` creates a a Keccak256 hash forced to be `02` and 30 zeroes...
At first I didn't realize this was the case and did not understand how this test could pass 😬